sycl/source/detail/scheduler/graph_builder.cpp

```diff
@@ -679,18 +709,32 @@ Scheduler::GraphBuilder::addCG(std::unique_ptr<detail::CG> CommandGroup,
 
     for (Command *Dep : Deps)
       NewCmd->addDep(DepDesc{Dep, Req, AllocaCmd});
+
+    if (CGType == CG::CGTYPE::CODEPLAY_HOST_TASK)
+      AllocasForReqs.push_back(AllocaCmd);
+  }
+
+  if (CGType == CG::CGTYPE::CODEPLAY_HOST_TASK) {
```

Comment:
Suggest moving changes in addCG to a separate function.

Reply:
Done.